android/Drawer/app/src/main/java/com/example/drawer/MQTTController.java

```diff
@@ -1,109 +1,165 @@
     private static MemoryPersistence persistence;
     private static MqttClient mqttClient;
+    private static final String TAG = "MainActivity";
+    private static final String STARTTAG = "Startup";
+    private static final String SUBTAG = "Subscription";
+    private static final String PUBTAG = "Publishing";
+    private static final String ETAG = "Error";
+    private static final HashMap<TextView, String> subscriptionMap = new HashMap<>();
 
     private MQTTController() {
         persistence = new MemoryPersistence();
     }
 
     public static void connect() {
         try {
+            //Create client
             mqttClient = new MqttClient(broker, clientId, persistence);
             MqttConnectOptions connOpts = new MqttConnectOptions();
             connOpts.setCleanSession(true);
 
-            System.out.println("Connecting to broker: " + broker);
+            //Attempt connection
+            Log.d(STARTTAG, "Connecting to broker: " + broker);
             mqttClient.connect(connOpts);
-            System.out.println("Connected");
+            if (!notConnected()) {
+                Log.d(STARTTAG, "Connected");
+            } else {
+                Log.d(ETAG, "Could not connect");
+            }
+
+            //Enable callback
+            mqttClient.setCallback(new MqttCallback() {
+                @Override
+                public void connectionLost(Throwable throwable) {
+                    Log.d(ETAG, "Connection lost");
+                    throwable.printStackTrace();
+                }
+
+                @Override
+                public void messageArrived(String s, MqttMessage mqttMessage) {
+                    String message = new String(mqttMessage.getPayload());
+                    Log.d(SUBTAG, "Topic:   " + s);
+                    Log.d(SUBTAG, "Message: " + message);
+                    // TODO optimize, not crucial due to small size of hashmap, but needed eventually. -MH
+                    // potentially make it topic keyed, then have lists of objects, could be good
+                    subscriptionMap.forEach((text, topic) -> {
+                        Log.d(SUBTAG, ("text: " + text));
+                        if (s.equals(topic)) text.setText(message);
+                    });
+                }
+
+                @Override
+                public void deliveryComplete(IMqttDeliveryToken token) {
+                    Log.d(PUBTAG, "Delivery complete.");
+                }
+            });
         } catch (MqttException e) {
             //Standard error printing
-            System.out.println("Could not connectd");
-            System.out.println("reason " + e.getReasonCode());
-            System.out.println("msg " + e.getMessage());
-            System.out.println("loc " + e.getLocalizedMessage());
-            System.out.println("cause " + e.getCause());
-            System.out.println("excep " + e);
+            Log.d(ETAG, "Could not connect");
```

Comment:
I understand that logging can be quite beneficial while debugging but the number of Log.d calls should be reduced here.

Reply:
Fixed in most recent commit.